lib/method_source.rb

```diff
@@ -49,8 +49,7 @@ def self.comment_helper(source_location, name=nil)
   # @return [Array<String>]  the contents of the file
   # @raise [SourceNotFoundError]
   def self.lines_for(file_name, name=nil)
-    @lines_for_file ||= {}
-    @lines_for_file[file_name] ||= File.readlines(file_name)
+    File.readlines(file_name)
```

Comment:
Ah good catch, but i wonder how badly this slows things down in tools that use this functionality (like Pry). 

Maybe it would be better to move to a helper object that does the memoization but which gets thrown away when it's no longer being used. Really silly of us (probably me) to put the memoization on the actual module itself!